### PR TITLE
Add mutation `def my_method; end` -> `remove_method :my_method`

### LINKED ISSUE #108: Add Mutation: def my_method; end -> remove_method :my_method
The purpose of this mutation is to remove methods that are duplicating functionality in their parent class. It will expose cases where methods were pushed up to the parent, but not removed from the child classes.

### PR DESCRIPTION
**Pull request by backus**
closes #108

### CODE REVIEW

**Comment by backus** [comment]
Currently failing due to the mutation issue I mentioned in #414.

**Comment by mbj** [comment]
@backus you wanna rebase because of changes in https://github.com/mbj/mutant/pull/418.

**Comment by backus** [comment]
Rebased

File: lib/mutant/mutator/node/define.rb [hunk 1]
```diff
@@ -25,6 +25,20 @@ class Instance < self
...
+            super()
+            emit_remove_method
+          end
+
+          # Remove an instance method
+          #
+          # @return [undefined]
+          #
+          # @api private
+          #
```
Comment (mbj) [inline lib/mutant/mutator/node/define.rb]: We do not place a newline between the `@api private` and the method anymore. See https://github.com/mbj/mutant/pull/364/files

Comment (backus) [inline lib/mutant/mutator/node/define.rb]: Whoops noted. Currently the default config of my YARD macro

**Comment by backus** [comment]
Fixed YARD doc and added a changelog entry

**Comment by mbj** [comment]
@backus Can you rebase this?

**Comment by backus** [comment]
Rebased @mbj

**Comment by mbj** [comment]
@backus Thx. Again!

**Comment by backus** [comment]
:smile:

**Pull request merged**